### DISCUSSION TITLE
Request for Java 11 support on TDP

### DISCUSSION THREAD

**Discussion by theblack-owl** [TOSIT-IO/TDP#86] [New features]
We are writing to request Java 11 compatibility for TDP. As a valued user of TDP, we greatly appreciate the continuous improvements and updates you provide, which enhance our productivity and help us meet our goals.

By not upgrading to a newer Java version, we may miss out on important bug fixes, security patches, and support from the Java community, leaving our software vulnerable to potential issues. This could hinder innovation, limit our ability to use new language features, and impact the competitiveness of our software in today's technology landscape.

Furthermore, newer versions of Spring Boot require a minimum Java version for compatibility. Staying on Java 8 prevents us from accessing the latest features, enhancements, and performance optimizations offered by Spring Boot updates. This restricts our ability to use newer Spring Boot modules and libraries that rely on higher Java versions. It also increases the risk of security vulnerabilities as we miss out on bug fixes and patches introduced in subsequent Java releases. Additionally, as the Java ecosystem evolves, third-party libraries and frameworks may stop supporting Java 8, making it difficult to integrate and maintain the application with the latest technologies.

Java 11 brings significant performance improvements, security updates, and new language features. Upgrading our Java runtime environment to Java 11 would enable us to take advantage of these benefits, optimize performance, and ensure the security of our application. It would also contribute to the long-term stability and maintainability of our software stack.

Thank you for considering our request, and we appreciate your attention to this matter.

**Comment by Pierrotws** [comment]
Hi,

For information, Java 11 will reach End of Life way before java 8.

```
|---------------------------------------|
|  version  | End of support |
|---------------------------------------|
|    java 8   |   2030 (Oracle)  |
|                  |  2026 (Red Hat)|
|---------------------------------------|
|  java 11   |   2026 (Oracle)  |
|                 |   2024 (Red hat)|
|---------------------------------------|
```
(source: https://en.wikipedia.org/wiki/Java_version_history)


Also, nothing prevent user to use higher java version for Spring project, or other client that will connect to the cluster.

The impact here could be spark, but using java 8 with Spark is very well supported.
Can you explain a bit more why java version on TDP cluster impacts application usage ?

Regards,

**Comment by Edouard-R** [comment]
Java 17 may be a better choice ^^
